File: notifications-polyglot/java/src/main/java/com/example/notification/controller/NotificationController.java
package com.example.notification.controller;

import com.example.notification.common.ApiResponse;
import com.example.notification.dto.NotificationRequest;
import com.example.notification.service.NotificationService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/notify")
@RequiredArgsConstructor
public class NotificationController {

    private final NotificationService notificationService;

    @PostMapping
    public ResponseEntity<ApiResponse<Map<String, String>>> sendNotification(@RequestBody NotificationRequest request) {
        String id = notificationService.enqueueNotification(request);
        return ResponseEntity.accepted()
                .body(ApiResponse.success(Map.of("notification_id", id, "status", "enqueued")));
    }
}

File: notifications-polyglot/kotlin/src/main/kotlin/com/example/notification/controller/NotificationController.kt
package com.example.notification.controller

import com.example.notification.common.ApiResponse
import com.example.notification.dto.NotificationRequest
import com.example.notification.service.NotificationService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/notify")
class NotificationController(
    private val notificationService: NotificationService
) {

    @PostMapping
    suspend fun sendNotification(@RequestBody request: NotificationRequest): ResponseEntity<ApiResponse<Map<String, String>>> {
        val id = notificationService.enqueueNotification(request)
        return ResponseEntity.accepted()
            .body(ApiResponse.success(mapOf("notification_id" to id, "status" to "enqueued")))
    }
}
